test/integration/auth/mongodb_aws.test.ts

```diff
@@ -250,9 +261,22 @@ describe('MONGODB-AWS', function () {
 
           expect(result).to.not.be.instanceOf(MongoServerError);
           expect(result).to.be.a('number');
+          expect(n).to.be.eql(1);
 
           expect(calledArguments).to.deep.equal(test.calledWith);
         });
+
+        it('creates one AWS provider per client', async function () {
+          await client.close();
+          await client.connect();
+          await client
+            .db('aws')
+            .collection('aws_test')
+            .estimatedDocumentCount()
+            .catch(error => error);
+
+          expect(n).to.be.eql(1);
+        });
```

Comment:
The testing AC on the ticket also asks for a test that asserts `fromNodeProviderChain` is only called once - can we add that test as well?

Reply:
I renamed the meaningless `n` counter to make it more descriptive :) The `numberOfFromNodeProviderChainCalls` verifies that the subbed `MongoDBAWS.credentialProvider` is called only once.